examples/golang/waku.go

```diff
@@ -0,0 +1,326 @@
+                       jsonWakuMessage,
+                       timeoutMs,
+                       (WakuCallBack) callback,
+                       NULL));
+	}
+
+	void cGoWakuRelaySubscribe(void* ctx, char* pubSubTopic) {
+
+		WAKU_CALL ( waku_relay_subscribe(ctx,
+							pubSubTopic,
+							(WakuCallBack) callback,
+							NULL) );
+	}
+
+	void cGoWakuRelayUnsubscribe(void* ctx, char* pubSubTopic) {
+
+		WAKU_CALL ( waku_relay_unsubscribe(ctx,
+							pubSubTopic,
+							(WakuCallBack) callback,
+							NULL) );
+	}
+
+	void cGoWakuConnect(void* ctx, char* peerMultiAddr, int timeoutMs) {
+		WAKU_CALL( waku_connect(ctx,
+						peerMultiAddr,
+						timeoutMs,
+						(WakuCallBack) callback,
+						NULL) );
+	}
+
+	void cGoWakuListenAddresses(void* ctx, void* myStr) {
+		WAKU_CALL (waku_listen_addresses(ctx, (WakuCallBack) callback, myStr) );
+	}
+
+*/
+import "C"
+
+import (
+	"time"
+	"fmt"
+	"unsafe"
+)
+
+func WakuNew(jsonConfig string) unsafe.Pointer {
+	return C.cGoWakuNew(C.CString(jsonConfig))
+}
+
+func WakuStart(ctx unsafe.Pointer) {
+	C.cGoWakuStart(ctx)
+}
+
+func WakuStop(ctx unsafe.Pointer) {
+	C.cGoWakuStop(ctx)
+}
+
+func WakuDestroy(ctx unsafe.Pointer) {
+	C.cGoWakuDestroy(ctx)
+}
+
+func WakuVersion(ctx unsafe.Pointer) string {
+	var str = C.allocMyString()
+	defer C.freeMyString(str)
+
+	C.cGoWakuVersion(ctx, str)
+
+	var version = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return version
+}
+
+//export MyEventCallback
+func MyEventCallback(callerRet C.int, msg *C.char, len C.size_t, userData unsafe.Pointer) {
+	fmt.Println("Event received:", C.GoStringN(msg, C.int(len)))
+}
+
+func WakuSetEventCallback(ctx unsafe.Pointer) {
+	var str = C.allocMyString()
+	// Notice that we are not releasing the `str` memory in this case
+	// because we leave it available to the upcoming events
+	C.cGoWakuSetEventCallback(ctx, str)
+}
+
+func WakuContentTopic(ctx unsafe.Pointer,
+					  appName string,
+					  appVersion int,
+					  contentTopicName string,
+					  encoding string) string {
+	var cAppName = C.CString(appName)
+	var cContentTopicName = C.CString(contentTopicName)
+	var cEncoding = C.CString(encoding)
+	var str = C.allocMyString()
+
+	defer C.free(unsafe.Pointer(cAppName))
+	defer C.free(unsafe.Pointer(cContentTopicName))
+	defer C.free(unsafe.Pointer(cEncoding))
+	defer C.freeMyString(str)
+
+	C.cGoWakuContentTopic(ctx, 
+						cAppName,
+						C.int(appVersion),
+						cContentTopicName,
+						cEncoding,
+						str)
+
+	var contentTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return contentTopic
+}
+
+func WakuPubsubTopic(ctx unsafe.Pointer, topicName string) string {
+	var cTopicName = C.CString(topicName)
+	var str = C.allocMyString()
+
+	defer C.free(unsafe.Pointer(cTopicName))
+	defer C.freeMyString(str)
+
+	C.cGoWakuPubsubTopic(ctx, cTopicName, str)
+
+	var pubsubTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return pubsubTopic
+}
```

Comment:
What does this do? Topic string into the shard format?

Reply:
> What does this do? Topic string into the shard format?

Yes, this function is aimed to simplify the creation of a well-formatter pubsubtopic.